frame/election-provider-support/src/lib.rs

```diff
@@ -662,16 +656,166 @@ impl<AccountId: IdentifierT, Accuracy: PerThing128, Balancing: Get<Option<Balanc
+pub struct ElectionBounds {
+	pub voters: DataProviderBounds,
+	pub targets: DataProviderBounds,
+}
+
+/// Utility builder for [`ElectionBounds`].
+#[derive(Copy, Clone)]
+pub struct ElectionBoundsBuilder {
+	voters: Option<DataProviderBounds>,
+	targets: Option<DataProviderBounds>,
+}
+
+impl ElectionBoundsBuilder {
+	/// Returns a new election bounds builder, initialized with unbounded voter and target limits.
+	pub fn new() -> Self {
+		ElectionBoundsBuilder { voters: None, targets: None }
+	}
+
+	/// Returns a new election bounds builder from an instance of `ElectionBounds`.
+	pub fn from(bounds: ElectionBounds) -> Self {
+		ElectionBoundsBuilder { voters: Some(bounds.voters), targets: Some(bounds.targets) }
+	}
+
+	// Sets the voters count bounds.
+	pub fn voters_count(mut self, count: CountBound) -> Self {
+		self.voters = self.voters.map_or(
+			Some(DataProviderBounds { count: Some(count), size: None }),
+			|mut bounds| {
+				bounds.count = Some(count);
+				Some(bounds)
+			},
+		);
+		self
+	}
+
+	// Sets the voters size bounds.
+	pub fn voters_size(mut self, size: SizeBound) -> Self {
+		self.voters = self.voters.map_or(
+			Some(DataProviderBounds { count: None, size: Some(size) }),
+			|mut bounds| {
+				bounds.size = Some(size);
+				Some(bounds)
+			},
+		);
+		self
+	}
+
+	// Sets the targets count bounds.
+	pub fn targets_count(mut self, count: CountBound) -> Self {
+		self.targets = self.targets.map_or(
+			Some(DataProviderBounds { count: Some(count), size: None }),
+			|mut bounds| {
+				bounds.count = Some(count);
+				Some(bounds)
+			},
+		);
+		self
+	}
+
+	// Sets the targets size bounds.
+	pub fn targets_size(mut self, size: SizeBound) -> Self {
+		self.targets = self.targets.map_or(
+			Some(DataProviderBounds { count: None, size: Some(size) }),
+			|mut bounds| {
+				bounds.size = Some(size);
+				Some(bounds)
+			},
+		);
+		self
+	}
+
+	/// Set the voters bounds.
+	pub fn voters(mut self, bounds: Option<DataProviderBounds>) -> Self {
+		self.voters = bounds;
+		self
+	}
+
+	/// Set the targets bounds.
+	pub fn targets(mut self, bounds: Option<DataProviderBounds>) -> Self {
+		self.targets = bounds;
+		self
+	}
+
+	/// Caps the maximum number of the voters bounds to `voters`. If `voters` bounds are less than
+	/// the current value, keeps it.
+	pub fn max_voters(mut self, voters: DataProviderBounds) -> Self {
+		self.voters = self.voters.map_or(None, |v| Some(v.max(voters)));
+		self
+	}
+
+	/// Caps the maximum number of the targets to `targets`. If `targets` bounds are less than the
+	/// current value, keeps it.
+	pub fn max_targets(mut self, targets: DataProviderBounds) -> Self {
+		self.targets = self.targets.map_or(None, |t| Some(t.max(targets)));
+		self
+	}
+
+	/// Returns an instance of `ElectionBounds` from the current state.
+	pub fn build(self) -> ElectionBounds {
+		ElectionBounds {
+			voters: self.voters.unwrap_or_default(),
+			targets: self.targets.unwrap_or_default(),
+		}
+	}
+}
```

Comment:
all of this could go a new file, re-exported from root, and have very minimal tests for the builder (if any of it has significant logic to it)